### PR TITLE
chore(docs): add authz docs

### PR DESCRIPTION
**Pull request by savonarola**

### CODE REVIEW

File: en_US/security/authz/authz.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    query = "SELECT permission, action, topic FROM acl WHERE username = ${username}"
+    server = "10.12.43.12:3306"
+}
+```
+
+## Authorization Chain
+
+Configured authorizers form a global chain. When a client makes a publish/subscribe request, authorizers are
+sequentially used to find access lists for the client. If an authorizer finds ACL rules, the request is checked
+against them and gets accepted/rejected. If ACL rules are not found, then the next authorizer from the chain is used.
```
Comment (zmstone) [inline en_US/security/authz/authz.md]:
```suggestion
against them and gets allowed/denied. If ACL rules are not found, then the next authorizer from the chain is used.
```

File: en_US/security/authz/authz.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+If no authorizer finds any ACL rules, then the default permission is applied.
+
+The first element of the chain is always the implicit authorizer that matches the request against the ACL rules
+returned by the authentication (if there are any).
```
Comment (zmstone) [inline en_US/security/authz/authz.md]: not sure I can understand this.
could you help to explain it?

File: en_US/security/authz/authz.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+## ACL Cache
+
+If a client sends requests intensively, it may be resource-consuming to fetch ACL rules for each request and match it
+against the rules. Therefore, ACL cache may be enabled to cache authorization results for a particular time.
+
+::: warning
+Caching improves performance significantly, so adjusting the default values to the relevant ones is essential.
```
Comment (zmstone) [inline en_US/security/authz/authz.md:69]:
```suggestion
Caching improves performance significantly, so adjusting the default values to the relevant ones is essential.
You can find more information about cache configuration below.
```

File: en_US/security/authz/authz.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+:::
+
+## Authorization placeholders
+
+Many authorizers allow using _placeholders_ in their configuration.
+Placeholders work as template variables filled with the corresponding
+client's information when performing authentication.
```
Comment (zmstone) [inline en_US/security/authz/authz.md]:
```suggestion
Placeholders work as template variables filled with the corresponding runtime information.

### Placeholders in configurations

Configuration templates can be filled with client information such ad client ID and user name at runtime.
```

File: en_US/security/authz/authz.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  enable = true
+  type = redis
+
+  ... other parameters ...
+
+  cmd = "HGETALL mqtt_user:${username}"
+}
+```
+
+## Topic placeholders
```
Comment (zmstone) [inline en_US/security/authz/authz.md]:
```suggestion
### Placeholders in ACL rules
```

File: en_US/security/authz/file.md [hunk 6]
```diff
@@ -1,24 +1,33 @@
...
+ACL configuration file is a list of Erlang [tuples](https://www.erlang.org/doc/reference_manual/data_types.html#tuple) ending with a period. A _tuple_ is a comma-separated list of expressions. The whole list is enclosed in curly braces.
 
-The built-in ACL is the lowest priority rule table. If it is not hit after all the ACL checks are completed, the default ACL rule is checked.
+The '%%` prefix identifies comment strings.
 
-The rules file is described in Erlang syntax:
+Example:
 
 ```erlang
 %% Allow "dashboard" users to subscribe to "$SYS/#" topics
```
Comment (zmstone) [inline en_US/security/authz/file.md]:
```suggestion
%% Allow MQTT client using username "dashboard"  to subscribe to "$SYS/#" topics
```

File: en_US/security/authz/file.md [hunk 7]
```diff
@@ -34,64 +43,42 @@ The rules file is described in Erlang syntax:
...
+Additionally, there are two special rules:
+    - `{allow, all}` — allow all operations.
+    - `{deny, all}` — deny all operations.
+
+These rules are usually used as default at the end of the configuration.
 
 ::: tip
-Only a few simple and general rules is contained in acl.conf that make it a system-based ACL principle. If you need to support complex, large amounts of ACL content, you should implement it in an authentication plugin.
-
+File-based authorization is most suitable for a small number of general rules. If you have many client-specific ACL rules
```
Comment (zmstone) [inline en_US/security/authz/file.md]: move this tip to the top

File: en_US/security/authz/file.md [hunk 8]
```diff
@@ -1,24 +1,33 @@
-# ACL file
+# File
 
-The built-in ACL sets rules through files, which is simple and lightweight to use. It is suitable for projects with predictable number of rules, no change or small change requirements.
+This authorizer implements ACL checks through matching pub/sub requests against a predefined list of rules configured in
+a file.
 
-ACL rules file:
+This is a simple and lightweight mechanism. It is suitable for projects with a limited number of rules that do not change
+frequently.
```
Comment (zmstone) [inline en_US/security/authz/file.md]: now acl file content can be posted from dashbaord at runtime.
```suggestion
File-based ACL is simple and lightweight. It is suitable to configure generic rules. For hundreds or more per-client rules, it is recommended to use other authorization sources, and file-based ACL can be the safety guard put at the end of the authorization chain.
```

File: en_US/security/authz/file.md [hunk 9]
```diff
@@ -1,24 +1,33 @@
-# ACL file
+# File
 
-The built-in ACL sets rules through files, which is simple and lightweight to use. It is suitable for projects with predictable number of rules, no change or small change requirements.
+This authorizer implements ACL checks through matching pub/sub requests against a predefined list of rules configured in
+a file.
```
Comment (zmstone) [inline en_US/security/authz/file.md:4]:
```suggestion
This authorizer implements ACL checks through matching pub/sub requests against a predefined list of rules configured in
a file.

::: tip Tip
Starting from 5.0, file-based ACL rules can be edited and reloaded from EMQX dashboard UI.
:::
```

File: en_US/security/authz/file.md (hunk 7 above)
Comment (savonarola) [reply en_US/security/authz/file.md]: Being on top, this tip now duplicates the similar paragraph, so I removed the tip.

File: en_US/security/authz/authz.md (hunk 2 above)
Comment (savonarola) [reply en_US/security/authz/authz.md]: I tried to simplify that and added a separate section.

Comment (savonarola) [reply en_US/security/authz/authz.md]: https://github.com/emqx/emqx-docs/pull/1216/commits/da09414d538babb98c5485be64969292ce731efe

**Review by zmstone — APPROVED** [review]

**Pull request merged**